libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/controller/AudioGuidanceStateController.kt

```diff
@@ -53,23 +47,23 @@ internal class AudioGuidanceStateController(
     override fun onAttached(mapboxNavigation: MapboxNavigation) {
         super.onAttached(mapboxNavigation)
 
-        val audioGuidanceApi = AudioGuidanceApi.create(mapboxNavigation, AudioGuidanceServices())
-        coroutineScope.launch {
-            flowSpeakInstructions().flatMapLatest { speakInstructions ->
-                if (speakInstructions) {
-                    audioGuidanceApi.speakVoiceInstructions()
+        val audioGuidance = MapboxNavigationApp.getObserver(MapboxAudioGuidance::class)
+        audioGuidance.stateFlow().observe {
+            if (it.isMuted != store.state.value.audio.isMuted) {
```

Comment:
Why do we need to create a new `AudioGuidanceState` inside `Store.state`. We could just reuse `MapboxAudioGuidanceState` inside `Store.state` and make `MapboxAudioGuidanceState` public. This way we don't need to map `MapboxAudioGuidanceState` to `AudioGuidanceState`. However, making it public would mean that we wouldn't be able to break `MapboxAudioGuidanceState` anytime we want.

We had the same problem with `RoadNameLabel` feature and we solved this issue by creating a factory. Look at `RoadFactory` that produces a `Road` object, which in turn is internal. Any other ideas are encouraged. But, I really think we don't need `AudioGuidanceState` as a separate file in `libnavui-dropin`

/cc @tomaszrybakiewicz

Reply:
> However, making it public would mean that we wouldn't be able to break MapboxAudioGuidanceState anytime we want.

I thought that the plan was to keep dropin ui AudioGuidanceState separate from libnavui-voice `MapboxAudioGuidance.State` because of that exact reason^.